src/ln/channel.rs

```diff
@@ -586,16 +590,40 @@ impl Channel {
 		if msg.max_accepted_htlcs > 483 {
 			return Err(ChannelError::Close("max_accpted_htlcs > 483"));
 		}
+		//optional parameter checking
+		// MAY fail the channel if
+		if msg.funding_satoshis < config.channel_limits.min_funding_satoshis {
+			return Err(ChannelError::Close("funding satoshis is less than the user specified limit"));
+		}
+		if msg.htlc_minimum_msat > config.channel_limits.max_htlc_minimum_msat {
+			return Err(ChannelError::Close("htlc minimum msat is higher than the user specified limit"));
+		}
+		if msg.max_htlc_value_in_flight_msat < config.channel_limits.min_max_htlc_value_in_flight_msat {
+			return Err(ChannelError::Close("max htlc value in flight msat is less than the user specified limit"));
+		}
+		if msg.channel_reserve_satoshis > config.channel_limits.max_channel_reserve_satoshis {
+			return Err(ChannelError::Close("channel reserve satoshis is higher than the user specified limit"));
+		}
+		if msg.max_accepted_htlcs < config.channel_limits.min_max_accepted_htlcs {
+			return Err(ChannelError::Close("max accepted htlcs is less than the user specified limit"));
+		}
+		if msg.dust_limit_satoshis < config.channel_limits.min_dust_limit_satoshis {
+			return Err(ChannelError::Close("dust limit satoshis is less than the user specified limit"));
+		}
 
 		// Convert things into internal flags and prep our state:
 
 		let their_announce = if (msg.channel_flags & 1) == 1 { true } else { false };
-		if require_announce && !their_announce {
-			return Err(ChannelError::Close("Peer tried to open unannounced channel, but we require public ones"));
-		}
-		if !allow_announce && their_announce {
-			return Err(ChannelError::Close("Peer tried to open announced channel, but we require private ones"));
+		if config.channel_limits.force_announced_channel_preference{
+			if local_config.announced_channel && !their_announce {
```

Comment:
nit: cant this just be local_config.announced_channel != their_announce? Do we really need specific error messages for each?

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.

Reply:
technically we dont, It might just make it easier to see why a channel was rejected.